File: library/src/test/java/me/proxer/library/api/messenger/SendMessageEndpointTest.java
package me.proxer.library.api.messenger;

import me.proxer.library.ProxerTest;
import me.proxer.library.api.ProxerException;
import me.proxer.library.enums.MessageAction;
import okhttp3.mockwebserver.MockResponse;
import org.junit.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

/**
 * @author Ruben Gees
 */
public class SendMessageEndpointTest extends ProxerTest {

    @Test
    public void testDefault() throws ProxerException, IOException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        final String result = api.messenger()
                .sendMessage("123", "message")
                .build()
                .execute();

        assertThat(result).isNull();
    }

    @Test
    public void testPath() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        api.messenger().sendMessage("123", "message")
                .build()
                .execute();

        assertThat(server.takeRequest().getPath())
                .isEqualTo("/api/v1/messenger/setmessage");
    }

    @Test
    public void testParameters() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        api.messenger().sendMessage("id", "someMessage")
                .build()
                .execute();

        assertThat(server.takeRequest().getBody().readUtf8())
                .isEqualTo("conference_id=id&text=someMessage");
    }

    @Test
    public void testActionParameters() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        api.messenger().sendMessage("id", MessageAction.REMOVE_USER, "Testerio")
                .build()
                .execute();

        assertThat(server.takeRequest().getBody().readUtf8())
                .isEqualTo("conference_id=id&text=%2FremoveUser%20Testerio");
    }

    @Test
    public void testInvalidAction() throws ProxerException, IOException, InterruptedException {
        //noinspection CodeBlock2Expr
        assertThatThrownBy(() -> {
            api.messenger().sendMessage("id", MessageAction.NONE, "")
                    .build()
                    .execute();
        }).isInstanceOf(IllegalArgumentException.class);
    }
}

File: library/src/test/kotlin/me/proxer/library/api/messenger/SendMessageEndpointTest.kt
package me.proxer.library.api.messenger

import me.proxer.library.ProxerTest
import me.proxer.library.enums.MessageAction
import me.proxer.library.runRequest
import org.amshove.kluent.invoking
import org.amshove.kluent.shouldBeNull
import org.amshove.kluent.shouldEqual
import org.amshove.kluent.shouldThrow
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class SendMessageEndpointTest : ProxerTest() {

    @Test
    fun testDefault() {
        val (result, _) = server.runRequest("empty.json") {
            api.messenger
                .sendMessage("123", "message")
                .build()
                .execute()
        }

        result.shouldBeNull()
    }

    @Test
    fun testPath() {
        val (_, request) = server.runRequest("empty.json") {
            api.messenger.sendMessage("123", "message")
                .build()
                .execute()
        }

        request.path shouldEqual "/api/v1/messenger/setmessage"
    }

    @Test
    fun testParameters() {
        val (_, request) = server.runRequest("empty.json") {
            api.messenger.sendMessage("id", "someMessage")
                .build()
                .execute()
        }

        request.body.readUtf8() shouldEqual "conference_id=id&text=someMessage"
    }

    @Test
    fun testActionParameters() {
        val (_, request) = server.runRequest("empty.json") {
            api.messenger.sendMessage("id", MessageAction.REMOVE_USER, "Testerio")
                .build()
                .execute()
        }

        request.body.readUtf8() shouldEqual "conference_id=id&text=%2FremoveUser%20Testerio"
    }

    @Test
    fun testInvalidAction() {
        invoking {
            api.messenger.sendMessage("id", MessageAction.NONE, "")
                .build()
                .execute()
        } shouldThrow IllegalArgumentException::class
    }
}
